docs/metrics/exemplars/Program.cs

```diff
@@ -0,0 +1,68 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using OpenTelemetry;
+using OpenTelemetry.Exporter;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Trace;
+
+public class Program
+{
+    private static readonly ActivitySource MyActivitySource = new("OpenTelemetry.Demo.Exemplar");
+    private static readonly Meter MyMeter = new("OpenTelemetry.Demo.Exemplar");
+    private static readonly Counter<long> MyFruitCounter = MyMeter.CreateCounter<long>("MyFruitCounter");
+
+    public static void Main()
+    {
+        var tracerProvider = Sdk.CreateTracerProviderBuilder()
+            .AddSource("OpenTelemetry.Demo.Exemplar")
+            .AddOtlpExporter()
+            .Build();
+
+        var meterProvider = Sdk.CreateMeterProviderBuilder()
+            .AddMeter("OpenTelemetry.Demo.Exemplar")
+            .SetExemplarFilter(ExemplarFilterType.TraceBased)
+            .AddOtlpExporter((exporterOptions, metricReaderOptions) =>
+            {
+                exporterOptions.Endpoint = new Uri("http://localhost:9090/api/v1/otlp/v1/metrics");
+                exporterOptions.Protocol = OtlpExportProtocol.HttpProtobuf;
+                metricReaderOptions.PeriodicExportingMetricReaderOptions.ExportIntervalMilliseconds = 1000;
+            })
+            .Build();
+
+        Console.WriteLine("Press any key to exit");
+
+        while (!Console.KeyAvailable)
+        {
+            using (var parent = MyActivitySource.StartActivity("Parent Operation"))
+            {
+                parent?.SetTag("key1", "value1");
+                parent?.SetTag("key2", "value2");
+
+                using (var child = MyActivitySource.StartActivity("Child Operation"))
+                {
+                    child?.SetTag("key3", "value3");
+                    child?.SetTag("key4", "value4");
+
+                    MyFruitCounter.Add(1, new("name", "apple"), new("color", "red"));
```

Comment:
I am not convinced that this is the best way to demonstrate Exemplars. An asp.net core app is very easy for most users to see value in exemplars, where they can navigate from Metric spikes to corresponding Traces, which is a common use case of Exemplars. 
The console app with manual spans do show what Exemplars would look like, but it is not very helpful to learn the idea of Exemplars.

Reply:
I want to focus on the basics, and for folks who want real world example, opentelemetry-demo is the right way to go IMHO.

Reply:
Resolved by https://github.com/open-telemetry/opentelemetry-dotnet/pull/5636#discussion_r1609028670.